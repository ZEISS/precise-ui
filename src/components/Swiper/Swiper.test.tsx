import * as React from 'react';
import * as enzyme from 'enzyme';
import { Swiper } from './';

describe('<Swiper />', () => {
  it('should render an empty <Swiper> component', () => {
    const wrapper = enzyme.shallow(<Swiper />);
    const nodes = wrapper.filter((node: any) => node.text() !== '');
    expect(nodes.exists()).toBeFalsy();
  });

  it('should not be controlled without props', () => {
    const wrapper = enzyme.shallow(
      <Swiper>
        <div>Content</div>
      </Swiper>,
    );
    expect(wrapper.state('controlled')).toBeFalsy();
  });

  it('should be controlled with selectedIndex', () => {
    const wrapper = enzyme.shallow(
      <Swiper selectedIndex={0}>
        <div>Content</div>
      </Swiper>,
    );
    expect(wrapper.state('controlled')).toBeTruthy();
  });

  it('should not be controlled with onPageChage', () => {
    const wrapper = enzyme.shallow(
      <Swiper onPageChange={() => {}}>
        <div>Content</div>
      </Swiper>,
    );
    expect(wrapper.state('controlled')).toBeFalsy();
  });

  it('should render a <Swiper> component with several pages', () => {
    const wrapper = enzyme.mount(
      <Swiper>
        <div>First page.</div>
        <div>Second page.</div>
        <div>Third page.</div>
        <div>Final page.</div>
      </Swiper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <Swiper> component with several pages in and selectedIndex', () => {
    const wrapper = enzyme.mount(
      <Swiper selectedIndex={1}>
        <div>First page.</div>
        <div>Second page.</div>
        <div>Final page.</div>
      </Swiper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
