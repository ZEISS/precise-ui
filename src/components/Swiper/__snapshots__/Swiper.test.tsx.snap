// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Swiper /> should render a <Swiper> component with several pages 1`] = `
<Swiper>
  <styled.div
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <div
      className="sc-htpNat jLPIAx"
      onKeyDown={[Function]}
      tabIndex={0}
    >
      <styled.div>
        <div
          className="sc-gzVnrw ehJJBp"
        >
          <InteractiveSurface
            onChange={[Function]}
            opaque={true}
          >
            <styled.div
              onClickCapture={[Function]}
              onMouseDownCapture={[Function]}
              onTouchStartCapture={[Function]}
            >
              <div
                className="sc-bdVaJa jAMQad"
                onClickCapture={[Function]}
                onMouseDownCapture={[Function]}
                onTouchStartCapture={[Function]}
              >
                <styled.div
                  innerRef={[Function]}
                  selectedIndex={0}
                >
                  <div
                    className="sc-bZQynM hKcHjN"
                  >
                    <styled.div
                      key="item-0"
                    >
                      <div
                        className="sc-EHOje cfvcOl"
                      >
                        <div>
                          First page.
                        </div>
                      </div>
                    </styled.div>
                    <styled.div
                      key="item-1"
                    >
                      <div
                        className="sc-EHOje cfvcOl"
                      >
                        <div>
                          Second page.
                        </div>
                      </div>
                    </styled.div>
                    <styled.div
                      key="item-2"
                    >
                      <div
                        className="sc-EHOje cfvcOl"
                      >
                        <div>
                          Third page.
                        </div>
                      </div>
                    </styled.div>
                    <styled.div
                      key="item-3"
                    >
                      <div
                        className="sc-EHOje cfvcOl"
                      >
                        <div>
                          Final page.
                        </div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </InteractiveSurface>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-bxivhb bcNCsA"
        >
          <styled.div
            active={true}
            index={0}
            key="bullet-0"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX bMqJke"
              onClick={[Function]}
            />
          </styled.div>
          <styled.div
            active={false}
            index={1}
            key="bullet-1"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX jdKdjT"
              onClick={[Function]}
            />
          </styled.div>
          <styled.div
            active={false}
            index={2}
            key="bullet-2"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX jdKdjT"
              onClick={[Function]}
            />
          </styled.div>
          <styled.div
            active={false}
            index={3}
            key="bullet-3"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX jdKdjT"
              onClick={[Function]}
            />
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Swiper>
`;

exports[`<Swiper /> should render a <Swiper> component with several pages in and selectedIndex 1`] = `
<Swiper
  selectedIndex={1}
>
  <styled.div
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <div
      className="sc-htpNat jLPIAx"
      onKeyDown={[Function]}
      tabIndex={0}
    >
      <styled.div>
        <div
          className="sc-gzVnrw ehJJBp"
        >
          <InteractiveSurface
            onChange={[Function]}
            opaque={true}
          >
            <styled.div
              onClickCapture={[Function]}
              onMouseDownCapture={[Function]}
              onTouchStartCapture={[Function]}
            >
              <div
                className="sc-bdVaJa jAMQad"
                onClickCapture={[Function]}
                onMouseDownCapture={[Function]}
                onTouchStartCapture={[Function]}
              >
                <styled.div
                  innerRef={[Function]}
                  selectedIndex={1}
                >
                  <div
                    className="sc-bZQynM cbSLgC"
                  >
                    <styled.div
                      key="item-0"
                    >
                      <div
                        className="sc-EHOje cfvcOl"
                      >
                        <div>
                          First page.
                        </div>
                      </div>
                    </styled.div>
                    <styled.div
                      key="item-1"
                    >
                      <div
                        className="sc-EHOje cfvcOl"
                      >
                        <div>
                          Second page.
                        </div>
                      </div>
                    </styled.div>
                    <styled.div
                      key="item-2"
                    >
                      <div
                        className="sc-EHOje cfvcOl"
                      >
                        <div>
                          Final page.
                        </div>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </InteractiveSurface>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-bxivhb bcNCsA"
        >
          <styled.div
            active={false}
            index={0}
            key="bullet-0"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX jdKdjT"
              onClick={[Function]}
            />
          </styled.div>
          <styled.div
            active={true}
            index={1}
            key="bullet-1"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX bMqJke"
              onClick={[Function]}
            />
          </styled.div>
          <styled.div
            active={false}
            index={2}
            key="bullet-2"
            onClick={[Function]}
          >
            <div
              className="sc-ifAKCX jdKdjT"
              onClick={[Function]}
            />
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Swiper>
`;
